src/uct/ib/mlx5/ib_mlx5.h

```diff
@@ -78,10 +78,10 @@
 #  define UCT_IB_MLX5_UAR_ALLOC_TYPE_WC 0x0
 #endif
 
-#if HAVE_DECL_MLX5DV_UAR_ALLOC_TYPE_NC
-#  define UCT_IB_MLX5_UAR_ALLOC_TYPE_NC MLX5DV_UAR_ALLOC_TYPE_NC
+#if HAVE_DECL_MLX5DV_UAR_ALLOC_TYPE_NC_DEDICATED
+#  define UCT_IB_MLX5_UAR_ALLOC_TYPE_NC_DEDICATED MLX5DV_UAR_ALLOC_TYPE_NC_DEDICATED
```

Comment:
I would keep the name of UCX constant without _DEDICATED

Reply:
keep the original HAVE_DECL_MLX5DV_UAR_ALLOC_TYPE_NC? why? also, UCT_IB_MLX5_UAR_ALLOC_TYPE_NC? or UCT_IB_MLX5_UAR_ALLOC_TYPE_NC_DEDICATED?

Reply:
in the context of mlx5dv macros (and HAVE_DECL_ derived from it) it's important to distinguish between nc and nc_dedicated. 
In the UCX context, we need proper NC const so in this case use of dedicated is only implementation detail

Reply:
renamed "UCT_IB_MLX5_UAR_ALLOC_TYPE_NC_DEDICATED" to "UCT_IB_MLX5_UAR_ALLOC_TYPE_NC"

Reply:
@Artemy-Mellanox is it okay now?